repronim/container/dockercontainer.py

```diff
@@ -1,156 +1,146 @@
-        """
-        Deletes a container from the Docker engine.
-        """
-        self.client.remove_container(self.container_id)
-
-    def remove_image(self):
-        """
-        Deletes an image from the Docker engine.
-        """
-        self.client.remove_container(self.image_id)
\ No newline at end of file
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the repronim package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Container sub-class to provide management of container engine."""
+
+from io import BytesIO
+import docker
+import json
+
+from repronim.container.base import Container
+
+
+class DockercontainerContainer(Container):
+    """
+    Container manager which talks to a Docker engine.
+    """
+
+    def __init__(self, resource, config = {}):
+        """
+        Class constructor
+
+        Parameters
+        ----------
+        resource : object
+            Resource sub-class instance
+        config : dictionary
+            Configuration parameters for the container.
+        """
+
+        self._image_id = None
+        self._container_id = None
+
+        super(DockercontainerContainer, self).__init__(resource, config)
+
+        # Initialize the client connection to Docker engine.
+        self._client = docker.Client(self.get_config('engine_url'))
+
+    def create(self, image_id=None):
+        """
+        Create a baseline Docker image and run it to create the container.
+
+        Parameters
+        ----------
+        image_id : string
+            Docker repository:tag string that identifies the Docker image
+            to use when creating the container. e.g. "debian:8.5"
+        """
+        if not image_id:
+            image_id = self.get_config('base_image_tag')
+
+        dockerfile = self._get_base_image_dockerfile(image_id)
+        self._build_image(dockerfile)
+        self._run_container()
+
+    def execute_command(self, command, env=None):
+        """
+        Execute the given command in the container.
+
+        Parameters
+        ----------
+        command : string or list
+            Shell command to send to the container to execute. The command can
+            be a string or a list of tokens that create the command.
+        env : dict
+            Additional (or replacement) environment variables which are applied
+            only to the current call
+
+        Returns
+        -------
+        list
+            List of STDOUT lines from the container.
+        """
+
+        command_env = self.get_updated_env(env)
+
+        if command_env:
+            # TODO: might not work - not tested it
+            command = ['%s=%s;' % k for k in command_env.items()] + command
```

Comment:
coverage says that it is covered ;)   
but I think you need to export those all so they propagate into child processes within, i.e `'export %s=%s'`.  Then also look out for those which might have spaces in them, so you might want to use `export %s=%r` ?